lib/fluent/plugin/in_s3.rb

```diff
@@ -209,12 +209,20 @@ def setup_credentials
         credentials_options[:port] = c.port if c.port
         credentials_options[:http_open_timeout] = c.http_open_timeout if c.http_open_timeout
         credentials_options[:http_read_timeout] = c.http_read_timeout if c.http_read_timeout
-        options[:credentials] = Aws::InstanceProfileCredentials.new(credentials_options)
+        if ENV["AWS_CONTAINER_CREDENTIALS_RELATIVE_URI"]
+          options[:credentials] = Aws::ECSCredentials.new(credentials_options)
+        else
+          options[:credentials] = Aws::InstanceProfileCredentials.new(credentials_options)
+        end
       when @shared_credentials
         c = @shared_credentials
         credentials_options[:path] = c.path if c.path
         credentials_options[:profile_name] = c.profile_name if c.profile_name
-        options[:credentials] = Aws::SharedCredentials.new(credentials_options)
+        if ENV["AWS_CONTAINER_CREDENTIALS_RELATIVE_URI"]
+          options[:credentials] = Aws::ECSCredentials.new(credentials_options)
+        else
+          options[:credentials] = Aws::InstanceProfileCredentials.new(credentials_options)
```

Comment:
why don't you use `Aws::SharedCredentials`?

Reply:
I opted to use the same mechanism that is currently used in the out_s3 module for consistency.

https://github.com/fluent/fluent-plugin-s3/blob/master/lib/fluent/plugin/out_s3.rb#L470

Reply:
https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/InstanceProfileCredentials.html
https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/ECSCredentials.html
https://github.com/fluent/fluent-plugin-s3/blob/1c9c7c8d13fd00aaf7307764f383028636300fe2/fluent-plugin-s3.gemspec#L20
https://github.com/fluent/fluent-plugin-s3/blob/1c9c7c8d13fd00aaf7307764f383028636300fe2/lib/fluent/plugin/in_s3.rb#L59-L64

`InstanceProfileCredentials` and `ECSCredentials`'s initialize method does not accept `path` and `path_name`.

Reply:
How about this point? From the docs, path and path_name are not supported. I'm not sure why this code works...

Reply:
Fair point; I don't think that line is necessary at all, I'll push a fix.